File: Pila.java
package practica3;

public class Pila {
    private long[] arreglo;
    private int top;
    private int n;

    public Pila(int n) {
        this.arreglo = new long[n];
        this.top = -1;
        this.n = n;
    }

    public void push(long e) {
        if (isFull()) {
            System.out.println("La pila está llena");
        } else {
            top++;
            arreglo[top] = e;
        }
    }

    public long pop() {
        if (isEmpty()) {
            System.out.println("La pila está vacía");
            return -1; // Valor de retorno indicativo de error
        } else {
            long e = arreglo[top];
            top--;
            return e;
        }
    }

    public long peek() {
        if (isEmpty()) {
            System.out.println("La pila está vacía");
            return -1; // Valor de retorno indicativo de error
        } else {
            return arreglo[top];
        }
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == n - 1;
    }

    // Ejemplo de uso
    public static void main(String[] args) {
        Pila pila = new Pila(5);
        pila.push(10);
        pila.push(20);
        System.out.println("Elemento en la cima: " + pila.peek());
        System.out.println("Eliminar elemento: " + pila.pop());
        System.out.println("Elemento en la cima: " + pila.peek());
    }
}

File: Pila.py
class Pila:
    def __init__(self, n):
        self.arreglo = [0] * n
        self.top = -1
        self.n = n

    def push(self, e):
        if self.isFull():
            print("La pila está llena")
        else:
            self.top += 1
            self.arreglo[self.top] = e

    def pop(self):
        if self.isEmpty():
            print("La pila está vacía")
            return None
        else:
            e = self.arreglo[self.top]
            self.top -= 1
            return e

    def peek(self):
        if self.isEmpty():
            print("La pila está vacía")
            return None
        else:
            return self.arreglo[self.top]

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.n - 1


# Ejemplo de uso
if __name__ == "__main__":
    pila = Pila(5)
    pila.push(10)
    pila.push(20)
    print("Elemento en la cima:", pila.peek())
    print("Eliminar elemento:", pila.pop())
    print("Elemento en la cima:", pila.peek())
